### PR TITLE
fix: proper source maps

### PR DESCRIPTION
**Pull request by lxsmnsyc**
- [x] Fix source maps for transformation processes.
- [ ] Fix source maps for client-transformed `"use server"` module
- [ ] Fix tests

### CODE REVIEW

File: packages/vinxi/lib/plugins/tree-shake.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import jsx from "@babel/plugin-syntax-jsx";
```
Comment (nksaraf) [inline packages/vinxi/lib/plugins/tree-shake.js:1]: I worry that when we remove the direct import, these plugins are not found in different package managers. I dont know if babel manages that internally well. Can we do the direct import? Does it cause problems

Comment (lxsmnsyc) [reply packages/vinxi/lib/plugins/tree-shake.js:1]: JSX itself is built into Babel. The problem is we don't need to leverage Babel to compile both JSX and TypeScript, we can delegate these to ESBuild.

Comment (nksaraf) [reply packages/vinxi/lib/plugins/tree-shake.js:1]: the syntax plugins are what you are adding in parserOpts.plugins too. The only difference if right now I import the syntax plugins and add them explicitly. Whereas the new method will make babel look for those syntax plugins.

Comment (lxsmnsyc) [reply packages/vinxi/lib/plugins/tree-shake.js:1]: Actually the specific difference is in the transformation output. Adding the syntax plugins will compile/transform these into their vanilla JS code, not preserve them. Babel already knows about the syntaxes, they are just disabled like a feature flag.

This kind of implementation is not new btw, kind of derived this from Preact and implemented this on all of my babel-related libraries

Comment (nksaraf) [reply packages/vinxi/lib/plugins/tree-shake.js:1]: ahhh okay. thats great then!

**Pull request merged**